pkgs/applications/networking/browsers/brave/default.nix

```diff
@@ -41,63 +40,61 @@
 , xorg
 , zlib
 , xdg-utils
-, wrapGAppsHook
+, snappy
+, libopus
+
+# command line arguments which are always set e.g "--disable-gpu"
 , commandLineArgs ? ""
+
+# Necessary for USB audio devices.
+, pulseSupport ? true, libpulseaudio ? null
+
+# For video acceleration via VA-API (--enable-features=VaapiVideoDecoder,VaapiVideoEncoder)
+, libvaSupport ? true, libva
+, enableVideoAcceleration ? libvaSupport
+
+# For Vulkan support (--enable-features=Vulkan)
+, vulkanSupport ? false, addOpenGLRunpath
+, enableVulkan ? vulkanSupport
 }:
 
-let
+with lib;
 
-rpath = lib.makeLibraryPath [
-  alsa-lib
-  at-spi2-atk
-  at-spi2-core
-  atk
-  cairo
-  cups
-  dbus
-  expat
-  fontconfig
-  freetype
-  gdk-pixbuf
-  glib
-  gtk3
-  libdrm
-  libpulseaudio
-  libX11
-  libxkbcommon
-  libXScrnSaver
-  libXcomposite
-  libXcursor
-  libXdamage
-  libXext
-  libXfixes
-  libXi
-  libXrandr
-  libXrender
-  libxshmfence
-  libXtst
-  libuuid
-  mesa
-  nspr
-  nss
-  pango
-  pipewire
-  udev
-  wayland
-  xdg-utils
-  xorg.libxcb
-  zlib
-];
+let
+  opusWithCustomModes = libopus.override {
+    withCustomModes = true;
+  };
```

Comment:
That should be in top-level so that we do not mix two versions. Also please keep the attr name.

Reply:
I have also followed what is done in `google-chrome` and `chromium`, so I assumed it would be okay. But even so, I don't really understand from your comment how I should change this part, sorry ~~"

Reply:
If you use libopus anywhere else instead of opusWithCustomModes you would have two different opus libraries which could lead to hard to debug errors.

Reply:
I feel like I'm not smart enough to understand the exact issues at play here (yes, I know that there can be things such as binary incompatibilities, but not quite sure what the blast radius would be here) so maybe I'll just drop that part of changes – I mimicked it from the Chrome packages on the assumption that Brave shares a lot of things with Chrome, but the package seemed to work okay without it so far, so maybe it's better to drop it to get the PR in faster and if it turns out it makes a difference, there always can be a follow-up PR.

Reply:
It makes a difference when you hunted down a stupid bug for 3 hours but yeah, its nice to have.

Reply:
I've just removed `libopus`.